### PR TITLE
Basic usage examples

### PR DESCRIPTION
**Pull request by dominikl**
**This PR only affects the examples, not the rOMERO-gateway code itself**

As the README.md points to the examples directory, I thought I should tidy them up a bit and also update them to the current state.

### CODE REVIEW

**Comment by jburel** [comment]
As discussed in the office. We may want to migrate those examples to the training-notebooks repo
Something to discuss

**Comment by joshmoore** [comment]
Are they being run by travis? If so, I can see keeping them with the code as a middle ground to integration tests. But yeah, in general we should discuss the strategy of what test/example-like code exists where.

**Comment by dominikl** [comment]
Is there an easy way to run the examples with travis? I don't want to clutter the examples with testing code, but keep them as simple as possible, so that one can just copy and paste them, e. g. use the inituitve `OMEROServer(host = "localhost", username = "xxx", password = "xxx")` constructor and not the 'obscure' `OMEROServer(credentialsFile = ...)` one.

I'd actually quite like to get that merged before trying to get the romero-gateway on CRAN. The more examples (additional to the jupyter notebook) I could point to the better.

**Comment by joshmoore** [comment]
If you change `localhost` to `omero` then the example should be able to connect to https://github.com/openmicroscopy/omero-test-infra/blob/master/docker-compose.yml#L8 The username and password can be created in your `test-data` script.

**Comment by manics** [comment]
I had a discussion with @jburel recently about making it easier to test Python notebooks. We thought modifying the OMERO gateway connect methods to optionally take the connection parameters from environment variables (in a similar manner to how [`idr.connection`](https://github.com/IDR/idr-py/blob/v0.3.0/idr/connections.py#L48) works) would make things easier.

**Comment by joshmoore** [comment]
In general, :+1: for having a new connection strategy, but untangling it from the IDR deploy is probably beyond this PR.

**Comment by dominikl** [comment]
Could this PR still be merged? For now it is an improvement of the examples. We could decide later what to do on the longer run, remove them (and refer to training notebooks) or add them to the travis process.

File: examples/AddData.R [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+server <- connect(server)
+
+# Load the image (can be any other OME object, too) we want to attach the
+# data to, in that case the image with ID=1
+imageId <- 1
+image <- loadObject(server, "ImageData", imageId)
+
+# Attach the dataframe to the image.
+# Stores the data as OMERO HDF5 table, see
+# https://docs.openmicroscopy.org/omero/5.4.7/developers/Tables.html
```
Comment (jburel) [inline examples/AddData.R]: https://docs.openmicroscopy.org/latest/omero/developers/Tables.html unless you want to pin to a specific version

File: examples/AddData.R [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+library(romero.gateway)
+
+# Load "Iris" example dataset
+library(datasets)
+data("iris")
+data.frame(iris)
+
+# Connect to server
+server <- OMEROServer(host = "localhost", username = "xxx", password = "xxx")
```
Comment (jburel) [inline examples/AddData.R]: you may want to something like
```
user_name = readline('Username: ')
user_password <- getPass::getPass('OMERO password: ')
server <- OMEROServer(host = 'localhost', username=user_name, password=user_password, port= as.integer(4064))
```
instead

File: examples/AddData.R [hunk 3]
```diff
@@ -0,0 +1,51 @@
+###
+# Add dataframes, CSV files, plots, etc. to OMERO
+###
+
```
Comment (jburel) [inline examples/AddData.R]: you may also want to add 
``options(warn=-1)
``

**Comment by jburel** [comment]
Ran the first example:
```
An object of class "Image"
Slot "server":
An object of class "OMEROServer"
Slot "host":
[1] "eel.openmicroscopy.org"

Slot "port":
[1] 4064

Slot "username":
[1] "user-1"

Slot "password":
[1] "xxxx"

Slot "credentialsFile":
character(0)

Slot "gateway":
[1] "Java-Object{omero.gateway.Gateway@6cc558c6}"

Slot "user":
[1] "Java-Object{omero.gateway.model.ExperimenterData (id=8)}"

Slot "ctx":
[1] "Java-Object{SecurityContext [groupID=9, experimenter=null, sudo=false, serverInformation=null, compression=0.85]}"


Slot "dataobject":
[1] "Java-Object{omero.gateway.model.ImageData (id=4600)}"
[1] "File ID of the Iris_Example_dataset.csv: 267626"
  Sepal.Length    Sepal.Width     Petal.Length    Petal.Width   
 Min.   :4.300   Min.   :2.000   Min.   :1.000   Min.   :0.100  
 1st Qu.:5.100   1st Qu.:2.800   1st Qu.:1.600   1st Qu.:0.300  
 Median :5.800   Median :3.000   Median :4.350   Median :1.300  
 Mean   :5.843   Mean   :3.057   Mean   :3.758   Mean   :1.199  
 3rd Qu.:6.400   3rd Qu.:3.300   3rd Qu.:5.100   3rd Qu.:1.800  
 Max.   :7.900   Max.   :4.400   Max.   :6.900   Max.   :2.500  
       Species  
 setosa    :50  
 versicolor:50  
 virginica :50  
                
                
                
png: 2

```

The password is printed out I have replaced by xxx in the output
This should be avoided

**Comment by jburel** [comment]
``GetData.R`` failed:

```
Error in .jcall("RJavaTools", "Ljava/lang/Object;", "invokeMethod", cl, : omero.ResourceError
    serverStackTrace = "ome.conditions.ResourceError: /repositories/OMERO-DEV-merge-deploy/Files/1 (No such file or directory)
                        	at ome.services.RawFileBean.read(RawFileBean.java:434)
                        	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                        	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
                        	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
```

**Comment by dominikl** [comment]
- Printing out password: That's because the method returns the Image object and as it's not captured in a variable it's printed out (with all it's content/references which included the server connnection). For now I can wrap that call into an `invisible` statement, that supresses the output. In future I should probably just not return anything from methods like `attachDataframe`.
- `GetData.R` issue: That's the point where the example tries to load the CSV file which was attached previously. The id (line 38 `fid <- 1`) has to be replaced by the real "File ID" of the attached CSV file (OMERO.web shows that in the tool tip). I can make that clearer in the example.

**Comment by jburel** [comment]
Thanks for the clarification. Sorry I forgot to change the file ID. I only change the image's one

**Comment by jburel** [comment]
Thanks
Everything looks good now

**Pull request merged**